text/0000-default-serializers-and-adapters.md

```diff
@@ -0,0 +1,53 @@
+- Start Date: 2019-07-27
+- Relevant Team(s): Ember Data
+- RFC PR: https://github.com/emberjs/rfcs/pull/522
+- Tracking:
+
+# Deprecate default Adapter and Serializer fallbacks
+
+## Summary
+As part of Project Trim, https://github.com/emberjs/data/issues/6166, this deprecates the fallback and default adapter and serializer across ember-data including:
+1. deprecate -default serializer fallback in Store.serializerFor
+2. deprecate Adapter.serializer and Adapter.defaultSerializer fallbacks (which currently default to -json-api.
+3. deprecate Store.defaultAdapter (-json-api) and the -json-api adapter fallback behavior
+4. deprecate Model.toJSON and support the scenario for quick debug-mode inspection via debugData() method which in DEBUG would print a similar object to the current toJSON method.
+
+## Motivation
+
+The adapter and serializer packages provide reference implementations and base classes that are not required for applications that implement their own following the required interfaces for adapters and serializers as defined in their respective base classes.  Deprecating them allows us to simplify the lookup pattern and remove automatic injection and registration of potentially unused classes.
+
+In addition to removing use of initializer injection, this takes a significant step toward simplifying the mental model for how to determine what adapter/serializer is in use. Removing the defaults forces app developers to be more cognizant about the type of application level concerns vs model-specific concerns; they will now need to explicitly define and use specific adapters/serializers. After this RFC lands, apps will always use an adapter/serializer you explicitly put into your application and the rule will be "the adapter matching modelName falling back to application".
+
+## Detailed design
+The injection of `-default` and `-json-api` serializers will be removed in the next major version (4.0). Since this changes some core assumptions we will deprecate the reliance on the existence of the defaults. All deprecation warnings will only be shown in Dev mode.
```

Comment:
could we link to this RFC / decision?

Reply:
is dev mode the development version of `ember build`? i.e. ember's default.

Reply:
> could we link to this RFC / decision?

I'm not sure what you're referring to? I linked to [Project Trim](https://github.com/emberjs/data/issues/6166) in the summary section. Is it my phrasing of 'this RFC lands'? In that case I'm talking about these deprecations.



> is dev mode the development version of `ember build`? i.e. ember's default.

Yes. I am just calling it out to make sure the deprecations don't show up in a production build.

Reply:
Ah, got it thanks. I had thought that was part of another RFC or something.